umbraco/Umbraco-CMS
# V15: Implement cache tags

Tags aren't entirely implemented in dotnet 9, even though the interfaces are  there, this will have to wait for 15.1
# Notes
- Adds tags for entries, at this moment, only `content` & `media` tags are used.
- Removes now obsolete repository methods (and some that weren't used at all)

More notes:
We could use tags for many more things, we could for example have a `published` / `draft` tag, to just be able to remove all published content from the cache. But the use case for this is not there yet.

We've also discussed having `AncestorOrSelf` tags (keys) in there, to be able to remove a cache entry when an ancestor gets unpublished, but this would required refactoring of our cache refresher logic.

# How to test
- Set a breakpoint in the DatabaseCache repository (Get), aka `DatabaseCacheRepository.GetContentSourceAsync`, and click the refresh memory cache button in settings PublishedCache dashboard, it should get out all your content again (thus have removed it from the cache)
